pkg/services/provisioning/datasources/datasources.go

```diff
@@ -230,7 +233,7 @@ func makeCreateCorrelationCommand(correlation map[string]any, SourceUID string,
 		}
 
 		// config.type is a deprecated place for this value. We will default it to "query" for legacy purposes but non-query correlations should have type outside of config
-		if config.Type != correlations.CorrelationType("query") {
+		if config.Type != "" && config.Type != correlations.CorrelationType("query") {
 			return correlations.CreateCorrelationCommand{}, correlations.ErrInvalidConfigType
```

Comment:
Would it make sense to change the error message in that error?

A the moment it's 

```
	ErrInvalidConfigType             = errors.New("correlation contains non default value in config.type")
```

Maybe it could say config.type is deprecated?

Reply:
Okay, I changed the error and added better instructions around how to fix it